warehouse/manage/views/__init__.py

```diff
@@ -334,6 +339,8 @@ def reverify_email(self):
             )
             if verify_email_ratelimit.test(self.request.user.id):
                 send_email_verification_email(self.request, (self.request.user, email))
+                for email in self.request.user.emails:
+                    send_new_email_added_email(self.request, (self.request.user, email))
```

Comment:
Similar thought here: sending a reverification request isn't the same thing as adding an email to an account, so I'm not sure if it makes sense to send this email here.